spec.bs

```diff
@@ -0,0 +1,473 @@
+      boolean skipObservation = false;
+    };
+
+    partial interface Document {
+        Promise&lt;sequence&lt;BrowsingTopic&gt;&gt; browsingTopics(optional BrowsingTopicsOptions options);
+    };
+  </pre>
+
+
+  The <dfn for="Document" method>browsingTopics(options)</dfn> method steps are:
+  1. Let |document| be the current document where the method is invoked on.
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps [=in parallel=]:
+    1. Let |topicsCallerContext| be a [=topics caller context=] with default initial field values.
+    1. Set |topicsCallerContext|'s [=topics caller context/caller origin=] to |document|'s [=Document/origin=].
+    1. Set |topicsCallerContext|'s [=topics caller context/top level context domain=] to |document|'s [=top-level browsing context=]'s [=active document=]'s {{Document/domain}}.
+    1. If any of the following is true:
+        - |document|'s [=Document/origin=] is an [=opaque origin=].
+        - [=this=] is not [=allowed to use=] the <code><a href="#browsing-topics-policy-controlled-feature">browsing-topics</a></code> feature.
+        - [=this=] is not [=allowed to use=] the <code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code> feature.
+        - The user preference setting disallows the access to topics from |topicsCallerContext|'s [=topics caller context/caller origin=] or |topicsCallerContext|'s [=topics caller context/top level context domain=].
+
+        then:
+          1. [=Queue a global task=] on the <dfn>browsing topics task source</dfn> given [=this=]'s [=relevant global object=] to [=reject=] |promise| with a "{{NotAllowedError}}" {{DOMException}}.
+          1. Abort these steps.
+    1. Let |topics| be the result from running the [=topics for caller=] algorithm, with |topicsCallerContext| as input.
+    1. If <var ignore=''>options</var>["{{BrowsingTopicsOptions/skipObservation}}"] is false:
+        1. Run the [=collect page topics calculation input data=] algorithm with |document| as input.
+        1. Run the [=collect topics caller origin=] algorithm with |document| and |topicsCallerContext|'s [=topics caller context/caller origin=] as input.
+    1. [=Queue a global task=] on the [=browsing topics task source=] given [=this=]'s [=relevant global object=] to perform the following steps:
+        1. [=Resolve=] |promise| with |topics|.
+  1. Return |promise|.
+
+</section>
+
+<section>
+  <h2 id="handle-topics-fetch-request-header">Handle fetch(&lt;url&gt;, {browsingTopics: true})</h2>
+  Topics can be send in the HTTP header for {{WindowOrWorkerGlobalScope/fetch()}} request. The response header for a topics related request can specify whether the caller should to be recorded.
+
+  <h3 id="browsing-topics-associated-with-request-header">browsingTopics associated with Request</h3>
+  A [=request=] has an associated <dfn for=request>browsingTopics</dfn> flag. Unless stated otherwise it is unset.
```

Comment:
We generally try to give these more descriptive names than the JavaScript API. Something like a "send browsing topics header boolean" would work best.

(We're also trying to move away from flags and towards booleans.)